### DISCUSSION TITLE
Signed/Public PDF URL from object storage returns 404 when sent to be processed via an API

### DISCUSSION THREAD

**Discussion by Kristijan994** [supabase/supabase#3730] [Questions] [answered]
Hey all,

I've encountered an issue that is sort of weird, I'm basically using an API to return the text from a PDF file.
The PDF file is stored in the supabase object storage, however, when I request the API with a signed or public URL the API cannot process the file URL and returns a 404 error. 
`(Can't download file with provided URL. (URL is not valid and returns 404 code)`

I'm using pdf.co API to be able to retrieve the text from the PDF file, but the URL is never valid.
I've also tested the API with other PDF URLs such as the firebase one and it worked.

Any solutions?

Thanks in advance!

**Comment by inian** [comment, marked as answer]
Can you share the exact URL you are trying to access? If it is sensitive please send it to support@supabase.io

Thanks

**Reply by Kristijan994** [reply]
Sure, here's a public sample: `https://eqsvdtceabpuzohyxlsf.supabase.in/storage/v1/object/public/doc-test/PDF-Sample.pdf`

**Reply by inian** [reply]
Is it showing a 404 for you @Kristijan994? I am able to view the PDF file.

**Reply by Kristijan994** [reply]
It's actually working in the browser, but it's not working and returns 404 on the API response when I'm using the URL to parse the PDF data.

**Reply by Kristijan994** [reply]
Here's the response from `https://api.pdf.co/v1/pdf/convert/to/json2` which works with a firebase PDF URL.

```
{
    "error": true,
    "status": 404,
    "message": "Can't download file with provided URL. (URL 'https://eqsvdtceabpuzohyxlsf.supabase.in/storage/v1/object/public/doc-test/PDF-Sample.pdf' is not valid and returns 404 code)",
    "errorCode": null,
    "remainingCredits": 104,
    "credits": 0
}
```

It's also crashing my own express.js PDF parser API, which also works with firebase PDF URL.

**Reply by inian** [reply]
Unfortunately, that seems to be an issue that you will need to raise with pdf.co. We do not send a 404 if the object exists in a public bucket and not able to replicate it via curl or a browser call

**Reply by Kristijan994** [reply]
All right, that's why I mentioned my own express.js PDF parser API. 
My local API returns a JSON for PDF URLs even without SSL, but it's not working and/or crashing with the ones from supabase.

**Reply by inian** [reply]
Can you share your a reproducible example with express server (or a node script or curl) that Supabase Storage is sending a 404 when you don't expect it to?

**Reply by Kristijan994** [reply]
Here's the response from the supabase PDF URL: 

![image](https://user-images.githubusercontent.com/37381126/139622080-4a5039f2-76e1-459a-b010-ca4359617c6e.png)

And here's the response from the firebase PDF URL:

![image](https://user-images.githubusercontent.com/37381126/139624240-33e2917a-fcc9-4939-b8ce-b6a3474ea491.png)

**Reply by Kristijan994** [reply]
It's a simple PDF parser API that works with every PDF URL that I've tested, except the supabase one.
Wondering if it's an issue with the supabase cloud storage?

**Reply by inian** [reply]
Could you share the code for this example? Without that, I am not able to reproduce the issue.

**Reply by Kristijan994** [reply]
Sure, no problem, let me send it elsewhere, where do you prefer?

**Reply by inian** [reply]
you can share a Github repo / gist with me or send the script to support@supabase.io

**Reply by Kristijan994** [reply]
Done, shared the repo with you.

**Reply by inian** [reply]
I see you are using pdf-parse but that api requires the pdf body directly and not the url. they [suggest using a different module](https://www.npmjs.com/package/pdf-parse#basic-usage---http) if you want to pass in the url directly. Or you can make a request to supabse storage via `axios` and pass the response body to pdf-parse

**Reply by Kristijan994** [reply]
Thank you @inian, the crawler module worked for supabase pdf urls. Problem solved.
